source/wp-content/themes/wporg-showcase-2022/src/archive-results-context/index.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Block Name: Archive Results Context
+ * Description: Displays context information for archive or search results.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\Theme\Showcase_2022\Archive_Results_Context;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Render the block content.
+ *
+ * @return string Returns the block markup.
+ */
+function render() {
+	global $wp_query;
+
+	if ( is_search() ) {
+		return sprintf(
+			/* translators: %1$s number of results; %2$s keyword. */
+			_n(
+				'<p>We found <b>%1$s</b> result for <b>%2$s</b></p>',
+				'<p>We found <b>%1$s</b> results for <b>%2$s</b></p>',
+				$wp_query->found_posts,
+				'wporg'
+			),
+			number_format_i18n( $wp_query->found_posts ),
+			esc_html( $wp_query->query['s'] )
+		);
+	}
+
+	if ( is_tag() ) {
+		return sprintf(
+			/* translators: %1$s number of results; %2$s tag/category. */
+			_n(
+				'<p>There is <b>%1$s</b> site tagged with <b>%2$s</b></p>',
+				'<p>There are <b>%1$s</b> sites tagged with <b>%2$s</b></p>',
+				$wp_query->found_posts,
+				'wporg'
+			),
+			number_format_i18n( $wp_query->found_posts ),
+			esc_html( get_queried_object()->name )
+		);
+	}
+
+	return sprintf(
```

Comment:
I'd recommend doing the same updates here that I added in https://github.com/WordPress/wporg-showcase-2022/pull/33 to support the various block `supports` properties, it will make working with this block in templates easier in the future.